### PR TITLE
Add PHP support with foundation for PHP libraries to load third parties

### LINKED ISSUE #39: Add PHP support
In order to allow the JSON files to be supported by PHP consumers (e.g. WordPress plugins, etc...), we should explore including PHP wrappers + composer files directly in this repo. Some thoughts:

- An ideal situation would be to have both PHP and JS files in the same repository, but to set up the build process so that only JS-relevant files get built to NPM and Composer/Packagist (I assume we'd also want to have this up somewhere in a PHP package registry?)
    - Having a clear separation (maybe via different directories) would it make it easier to grok the codebase as well
- We'll have to do some testing to make sure no breaking changes are introduced to the current NPM dep

### PR DESCRIPTION
**Pull request by felixarntz**
Fixes #39 

* [x] PHP classes foundation
* [x] PHP third parties implementation
* [x] Test coverage
* [x] Add docs
* [ ] ~~Move JSON files out of JS-specific folders~~ (will be done in a follow up PR)

### CODE REVIEW

File: inc/Data/ThirdPartyDataFormatter.php [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+ */
+class ThirdPartyDataFormatter
+{
+
+    /**
+     * Formats third party data for a given set of input arguments and returns the corresponding output.
+     *
+     * @see https://github.com/GoogleChromeLabs/third-party-capital/blob/0831b937a8468e0f74bd79edd5a59fa8b2e6e763/src/utils/index.ts#L94
+     *
+     * @param ThirdPartyData $data Third party data to format.
```
Comment (adamsilverstein) [inline inc/Data/ThirdPartyDataFormatter.php:25]: align $data with line below

**Comment by adamsilverstein** [comment]
@felixarntz - this looks great so far! the main possible improvement would be adding tests although I'm not sure that should be a blocker since the JS side doesn't have them :)

**Comment by flashdesignory** [comment]
> @felixarntz - this looks great so far! the main possible improvement would be adding tests although I'm not sure that should be a blocker since the JS side doesn't have them :)

It's always great to have a new set of eyes on the JS side as well 😄 
We do cover the crucial functions with unit tests (utils file), but please let us know if you feel that there are some tests missing. 
Personally, I don't think we need 100% test coverage, but should cover the crucial parts to ensure we're delivering the correct output.

**Comment by felixarntz** [comment]
@housseindjirdeh @adamsilverstein @flashdesignory I've completed the remaining tests so this is now ready for full review.

Regarding documentation, for now I only added a small note to the readme referring to both the JS and PHP implementations. More wasn't really needed at this point, as everything else in the readme applies in both scenarios.

I think it would be useful to add more specific documentation (most importantly a few code examples in both JS and PHP), but that could be done in a separate PR.

File: inc/Data/ThirdPartyDataFormatter.php [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+     * @see https://github.com/GoogleChromeLabs/third-party-capital/blob/0831b937a8468e0f74bd79edd5a59fa8b2e6e763/src/utils/index.ts#L28
+     *
+     * @param string   $url          Base URL.
+     * @param string[] $params       Parameter names.
+     * @param array    $args         Input arguments for the src attribute query parameters.
+     * @param array    $slugParamArg Optional. Input argument for the src attribute slug query parameter.
+     *                               Default empty array.
+     * @return string HTML string.
+     */
+    public static function formatUrl(string $url, array $params, array $args, array $slugParamArg = array()): string
```
[the remark this answers is not in the document]
Comment (felixarntz) [reply inc/Data/ThirdPartyDataFormatter.php]: @adamsilverstein Added comprehensive test coverage for this class in https://github.com/GoogleChromeLabs/third-party-capital/pull/40/commits/c8c947d20997751f9806d6f679fafa30ac20c9c2.

**Review by housseindjirdeh — APPROVED** [review]

**Pull request merged**